bundles/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/DSMRPort.java

```diff
@@ -0,0 +1,350 @@
+ * method encounters problems it will automatically close itselves
+ * <p>
+ * The close method can be called asynchronous and will release OS resources.
+ * <p>
+ * In this way the DSMR port can restore the connection automatically
+ * <p>
+ * <code>
+ * An example DSMR telegram in accordance to IEC 62056-21 Mode D.<br>
+ * /ISk5\2MT382-1000<br>
+ * 0-0:96.1.1(4B384547303034303436333935353037)<br>
+ * 1-0:1.8.1(12345.678*kWh)<br>
+ * 1-0:1.8.2(12345.678*kWh)<br>
+ * 1-0:2.8.1(12345.678*kWh)<br>
+ * 1-0:2.8.2(12345.678*kWh)<br>
+ * 0-0:96.14.0(0002)<br>
+ * 1-0:1.7.0(001.19*kW)<br>
+ * 1-0:2.7.0(000.00*kW)<br>
+ * 0-0:17.0.0(016*A)<br>
+ * 0-0:96.3.10(1)<br>
+ * 0-0:96.13.1(303132333435363738)<br>
+ * 0-0:96.13.0(303132333435363738393A3B3C3D3E3F303132333435363738393A3B3C3D3E3F<br>
+ * 303132333435363738393A3B3C3D3E3F303132333435363738393A3B3C3D3E3F<br>
+ * 303132333435363738393A3B3C3D3E3F)<br>
+ * 0-1:96.1.0(3232323241424344313233343536373839)<br>
+ * 0-1:24.1.0(03)<br>
+ * 0-1:24.3.0(090212160000)(00)(60)(1)(0-1:24.2.1)(m3)<br>
+ * (00000.000)<br>
+ * 0-1:24.4.0(1)<br>
+ * !<br>
+ * </code>  
+ * @author M. Volaart
+ * @since 1.7.0
+ *         
+ */
+public class DSMRPort {
+	/* Internal state based on DSMR specification */
+	private enum ReadState {
+		WAIT_FOR_START, IDENTIFICATION, DATA, END
+	};
+
+	/* logger */
+	private static final Logger logger = LoggerFactory
+			.getLogger(DSMRPort.class);
+
+	/* private object variables */
+	private final String portName;
+	private final DSMRVersion version;
+	private final int timeoutMSec;
+	
+	/* serial port resources */
+	private SerialPort serialPort;
+	private BufferedReader reader;
+	
+	/* state variables */
+	private ReadState readerState;
+	private boolean isOpen = false;
+	
+	/* helpers */
+	private OBISMsgFactory factory;
+	
+	/* 
+	 * The portLock is used for the shared data used when opening and closing the port.
+	 * The following shared data must be guarded by the lock:
+	 * SerialPort, BufferedReader, isOpen
+	 */
+	private Object portLock = new Object();
+	
+	/**
+	 * Creates a new DSMRPort. This is only a reference to a port.
+	 * The port will not be opened nor it is checked if the DSMR Port can
+	 * successfully be opened.
+	 * 
+	 * @param portName Device identifier of the post (e.g. /dev/ttyUSB0)
+	 * @param version Version of the DSMR Specification. See {@link DSMRVersion}
+	 * @param timeoutMSec communication timeout in milliseconds
+	 */
+	public DSMRPort(String portName, DSMRVersion version, int timeoutMSec) {
+		this.portName = portName;
+		this.version = version;
+		this.timeoutMSec = timeoutMSec;
+
+		factory = new OBISMsgFactory(version);
+	}
+	
+	/**
+	 * Returns whether or not the port is open
+	 * 
+	 * @return true if the DSMRPort is open, false otherwise
+	 */
+	public boolean isOpen() {
+		synchronized(portLock) {
+			return isOpen;
+		}
+	}
+
+	/**
+	 * Closes the DSMRPort and release OS resources
+	 */
+	public void close() {
+		closePort();
+	}
```

Comment:
is this method meant to be enhancement somehow? If yes, please give an idea what else should be done here. If not, please remove this delegation to closePort().